### ISSUE TITLE
Should sidestep::Opcode flag fields be unsigned?

### ISSUE THREAD

**Issue by jtv** [gperftools/gperftools#848]
Cross-compiling 2.5 for Windows on x86-64 with gcc 6.2.0 gives me these errors:

```
src/windows/ia32_opcode_map.cc:343:1: error: narrowing conversion of ‘2148073472u’ from ‘unsigned int’ to ‘int’ inside { } [-Wnarrowing]
 };
 ^
src/windows/ia32_opcode_map.cc:947:1: error: narrowing conversion of ‘2148073472u’ from ‘unsigned int’ to ‘int’ inside { } [-Wnarrowing]
 };
 ^
```

The reason is that `OT_ADDRESS_MODE_M` is defined as `0x80000000` — too wide to fit into a 32-bit signed int — yet used to compute values for `sidestep::Opcode::flag_dest_`.  Something similar happens for one of the other flag fields.  The compiler is probably allowed to do surprising and horrible things in optimising this.

This diff makes it compile cleanly for me (inline, since github won't let me attach it):

```diff
diff -rc gperftools-2.5/src/windows/mini_disassembler_types.h gperftools-2.5-patched/src/windows/mini_disassembler_types.h
*** gperftools-2.5/src/windows/mini_disassembler_types.h	2015-10-01 17:02:57.000000000 +0200
--- gperftools-2.5-patched/src/windows/mini_disassembler_types.h	2016-11-25 16:08:24.512717045 +0100
***************
*** 188,196 ****
    // Description of the type of the dest, src and aux operands,
    // put together from an enOperandType flag and an enAddressingMethod
    // flag.
!   int flag_dest_;
!   int flag_source_;
!   int flag_aux_;
  
    // We indicate the mnemonic for debugging purposes
    const char* mnemonic_;
--- 188,196 ----
    // Description of the type of the dest, src and aux operands,
    // put together from an enOperandType flag and an enAddressingMethod
    // flag.
!   unsigned int flag_dest_;
!   unsigned int flag_source_;
!   unsigned int flag_aux_;
  
    // We indicate the mnemonic for debugging purposes
    const char* mnemonic_;
```

**Comment by alk** [comment]
Thanks, but then few other places that process those fields need to be converted to unsigned. Can you also please do it?

**Comment by jtv** [comment]
To be honest, I'm not confident that I understand these flags and their usage well enough!  I was hoping that somebody else might know.

**Comment by alk** [comment]
ah, this is actually fixed in 22f7ceb97a63bcbc9d824a80916bb7a809fad1f2
